hazelcast-cloud/src/main/java/com/hazelcast/aws/AWSClient.java

```diff
@@ -40,7 +42,7 @@ public AWSClient(AwsConfig awsConfig) {
         this.awsConfig = awsConfig;
         endpoint = awsConfig.getHostHeader();
         if (awsConfig.getRegion() != null && awsConfig.getRegion().length() > 0) {
-            setEndpoint("ec2." + awsConfig.getRegion() + ".amazonaws.com");
+            setEndpoint(awsConfig.getHostHeader().replace("ec2.", "ec2." + awsConfig.getRegion() + "."));
```

Comment:
user should provide a host-header starting with `ec2.`
because all headers start with `ec2.` http://docs.aws.amazon.com/general/latest/gr/rande.html#ec2_region
maybe we can make fail fast. what do you think?